### ISSUE TITLE
Timestream move Time from CommonAttributes to Records

### ISSUE THREAD

**Issue by araddas** [influxdata/telegraf#8848]
### Proposal:
Consider moving the Time Attribute to each Record, this would allow multiple interval checks to be sent in a single write. 

### Current behavior:
```
2021-02-11T22:18:23Z D! [outputs.timestream] Writing to Timestream: '{
  CommonAttributes: {
    Dimensions: [{
        Name: "host",
        Value: "test-timestream"
      },{
        Name: "name",
        Value: "sda"
      }],
    Time: "1613081796",
    TimeUnit: "SECONDS"
  },
  DatabaseName: "ssdev-or-tspoc-telegraf-db",
  Records: [
    {
      MeasureName: "write_bytes",
      MeasureValue: "887197743104",
      MeasureValueType: "BIGINT"
    },
    {
      MeasureName: "read_bytes",
      MeasureValue: "887197743104",
      MeasureValueType: "BIGINT"
    },
  ]
```
### Desired behavior:
```
2021-02-11T22:18:23Z D! [outputs.timestream] Writing to Timestream: '{
  CommonAttributes: {
    Dimensions: [{
        Name: "host",
        Value: "test-timestream"
      },{
        Name: "name",
        Value: "sda"
      }],
    TimeUnit: "SECONDS"
  },
  DatabaseName: "ssdev-or-tspoc-telegraf-db",
  Records: [
    {
      MeasureName: "write_bytes",
      MeasureValue: "887197743104",
      MeasureValueType: "BIGINT",
      Time: "1613091796"
    },
    {
      MeasureName: "read_bytes",
      MeasureValue: "887197743104",
      MeasureValueType: "BIGINT"
      Time: "1613091796",
    },
    {
      MeasureName: "write_bytes",
      MeasureValue: "887197743104",
      MeasureValueType: "BIGINT"
      Time: "1613081796",
    },
    {
      MeasureName: "read_bytes",
      MeasureValue: "887197743104",
      MeasureValueType: "BIGINT"
      Time: "1613081796",
    },
  ]
```
### Use case: 
By moving the Time Attribute into the Records it would allow multiple intervals to be sent in the same write (Given the flush interval is increased), this could reduce the amount of writes by 2x, 3x depending on the size of the dimensions. In turn this would save 2x, 3x in Timestream Write costs.

**Comment by araddas** [comment]
> Our Timestream testing has shown that the Timestream output can't even remotely come close to keeping up with other outputs, such as InfluxDB. We get `...did not complete within its flush interval` and `Metric buffer overflow; 800 metrics have been dropped` repeatedly. This is happening in our QA environment, suggesting when we get to production we will need about 100 of these writing to keep up.
> 
> This write-coalesce will help improve performance overall, which this plugin needs.

This pull request (https://github.com/influxdata/telegraf/pull/8947) resolves this issue, not sure why its taking so long for it to get approved.

**Comment by MyaLongmire** [comment]
@araddas @sysadmin1139 have you tested PR #8947 to ensure it works? If so do you have any feedback? We are currently putting through our review process. Thank you for your patience.

**Comment by araddas** [comment]
> @araddas @sysadmin1139 have you tested PR #8947 to ensure it works? If so do you have any feedback? We are currently putting through our review process. Thank you for your patience.

@MyaLongmire Yes I have, it was provided to me by our AWS account team and I provided them feedback in March 2021. The performance issues are resolved, however, the binary provided to me was compiled on Debian and I was running it on CentOS 7. Due to this a few of the input plugins broke, one in particular was the x509_cert plugin. I am sure that if the binary was compiled on a CentOS/Redhat/OEL box it would not have broken the input plugins. Below are the performance findings between the current output plugin and the fix in PR #8947.

### Current Output Plugin
```
[telegraf.d]# grep "Wrote batch" /var/log/telegraf/telegraf.log
2021-02-16T18:04:13Z D! [outputs.timestream] Wrote batch of 738 metrics in 1m17.381684011s
2021-02-16T18:06:14Z D! [outputs.timestream] Wrote batch of 737 metrics in 1m15.076720975s
2021-02-16T18:08:17Z D! [outputs.timestream] Wrote batch of 738 metrics in 1m16.568463066s
2021-02-16T18:10:21Z D! [outputs.timestream] Wrote batch of 772 metrics in 1m18.626007026s
2021-02-16T18:12:13Z D! [outputs.timestream] Wrote batch of 677 metrics in 1m7.5071232s
```

### PR #8947
```
[telegraf.d]# grep "Wrote batch" /var/log/telegraf/telegraf.log
2021-02-25T04:10:38Z D! [outputs.timestream] *Wrote batch* of 2369 metrics in 1.309035188s
2021-02-25T04:12:42Z D! [outputs.timestream] *Wrote batch* of 2413 metrics in 952.00412ms
2021-02-25T04:14:44Z D! [outputs.timestream] *Wrote batch* of 2409 metrics in 964.837444ms
2021-02-25T04:16:46Z D! [outputs.timestream] *Wrote batch* of 2393 metrics in 1.026671739s
2021-02-25T04:18:47Z D! [outputs.timestream] *Wrote batch* of 2574 metrics in 946.669363ms
2021-02-25T04:20:52Z D! [outputs.timestream] *Wrote batch* of 2387 metrics in 1.254998499s
2021-02-25T04:22:53Z D! [outputs.timestream] *Wrote batch* of 2414 metrics in 920.610652ms
2021-02-25T04:25:09Z D! [outputs.timestream] *Wrote batch* of 2390 metrics in 1.219812232s```
